src/metabase/search/postgres/ingestion.clj

```diff
@@ -100,25 +108,43 @@
   "Force ingestion to happen immediately, on the same thread."
   false)
 
-(defmacro ^:private run-on-thread [& body]
-  `(if *force-sync*
-     (do ~@body)
-     (doto (Thread. ^Runnable (fn [] ~@body))
-       (.start))))
+(defn- bulk-ingest! [updates]
+  (->> (for [[search-model where-clauses] (u/group-by first second updates)]
+         (spec-index-reducible search-model (into [:or] (distinct where-clauses))))
+       ;; init collection is only for clj-kondo, as we know that the list is non-empty
+       (reduce u/rconcat [])
+       (batch-update!)))
+
+(defn process-next-batch
+  "Wait up to 'delay-ms' for a queued batch to become ready, and process the batch if we get one.
+  Returns the number of search index entries that get updated as a result."
+  [first-delay-ms next-delay-ms]
+  (if-let [queued-updates (queue/take-delayed-batch! queue batch-max first-delay-ms next-delay-ms)]
+    (bulk-ingest! queued-updates)
+    0))
+
+(defn- index-worker-exists? []
+  (task/job-exists? @(requiring-resolve 'metabase.task.search-index/job-key-full))
```

Comment:
Oops, this line w for debugging and should be removed (just waiting for CI to run first)

Reply:
```suggestion
```